File: QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author jo_se
 */
public abstract class Casilla {

    private int numeroCasilla;
    private int coste;   
    private TipoCasilla tipo;

    //construtor para casillas que no son calle
    public Casilla(int numeroCasilla, TipoCasilla tipo) {
        this.numeroCasilla = numeroCasilla;
        this.tipo = tipo;
        this.coste = 0;
    }


    
    //Contructor para casillas que son impuesto
    public Casilla(int numeroCasilla, int coste, TipoCasilla tipo) {
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
        this.tipo = tipo;
        
    }
    
    
    public int getNumeroCasilla() {
        return numeroCasilla;
    }

    int getCoste() {
        return coste;
    }


    public TipoCasilla getTipo() {
        return tipo;
    }


    @Override
    public String toString() {
        String stringCasilla = "Casilla{\nNumero Casilla=" + numeroCasilla + "\nTipo="
                + tipo + "\nCoste=" + coste;
        /*if (tipo == TipoCasilla.CALLE) {
            stringCasilla += "\nNumero Casas=" + numCasas
                    + "\nNumero Hoteles=" + numHoteles + "\n" + titulo.toString();
        }*/

        stringCasilla += "\n}";

        return stringCasilla;
    }

    int precioTotalComprar() {
        throw new UnsupportedOperationException("sin implementar");
    }

    boolean calcularHipoteca() {
        throw new UnsupportedOperationException("sin implementar");
    }

    abstract boolean soyEdificable(); 
    public abstract TituloPropiedad getTitulo();
   
}

File: Qytetet_Ruby/lib/casilla.rb
# encoding: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  class Casilla
    
    #Atributos de la clase
    attr_reader :numeroCasilla, :coste, :tipo
    
    def initialize(numeroCasilla, coste, tipo)
      @numeroCasilla = numeroCasilla
      @coste = coste
      @tipo = tipo

    end
    

    def self.crear_impuesto(numeroCasilla, coste, tipo)
      self.new(numeroCasilla, coste,tipo)
    end
    
    #construtor para casillas que no son calle
    def self.crear_no_calle(numeroCasilla, tipo)
      self.new(numeroCasilla, nil, tipo)
    end
    
    def soy_edificable()
      return (@tipo == TipoCasilla::CALLE)
    end
    
    #Devuelve un string con la informacion de la casill
    def to_s
      mensaje = "\nNumero casilla #{@numeroCasilla}\nTipo #{@tipo}\n"
      
      if(tipo == TipoCasilla::CALLE)
        mensaje += "\nCoste #{@coste}\nNumero hoteles #{@numHoteles}\nNumero casas #{@numCasas}\nTitulo = " +@titulo.to_s()
      end
      
      return mensaje
    end   

  end
end
